### PR TITLE
Bugfix/2159 create tables

### PR DESCRIPTION
**Pull request by agneym**
Solves #2159.

Related to #2133

On Knex adapter, tables where not being created when `dropDatabase` was not enabled and hence the ToDo starter could only be run with `dropDatabase: true` for the first time. 

This PR calls `createTables` on each `postConnect` but creates tables only when they do not already exist. This should make the starter Postgres adapters run smoothly without the `dropDatabase` option.

### CODE REVIEW

File: packages/adapter-knex/lib/adapter-knex.js [hunk 1]
```diff
@@ -78,8 +78,6 @@ class KnexAdapter extends BaseKeystoneAdapter {
         console.log('Knex adapter: Dropping database');
       }
       await this.dropDatabase();
-    } else {
-      return [];
     }
     return this._createTables();
```
Comment (MadeByMike) [inline packages/adapter-knex/lib/adapter-knex.js]: Am I wrong in reading that this return statement will never be hit?

File: packages/adapter-knex/lib/adapter-knex.js [hunk 2]
```diff
@@ -282,9 +280,12 @@ class KnexListAdapter extends BaseListAdapter {
 
   async createTable() {
     // Let the field adapter add what it needs to the table schema
-    await this._schema().createTable(this.tableName, table => {
-      this.fieldAdapters.forEach(adapter => adapter.addToTableSchema(table));
-    });
+    const exists = await this._schema().hasTable(this.tableName);
```
Comment (MadeByMike) [inline packages/adapter-knex/lib/adapter-knex.js]: @timleslie thoughts on this?

File: packages/adapter-knex/lib/adapter-knex.js (hunk 1 above)
Comment (agneym) [reply packages/adapter-knex/lib/adapter-knex.js]: With the `return` on 81 and 82 this was indeed the case. The `return` for `createTables` was never being hit and hence the issue where no tables where created for Postgres people.

**Comment by raulrpearson** [comment]
I'm really looking forward to this PR being merged. I've been tearing my hair out for several days thinking I was [doing something wrong on deployment](https://github.com/keystonejs/keystone/issues/2488#issuecomment-596985380), but I think the culprit was the problem addressed in this PR. I'm also using the knex adapter with Postgres.

**Comment by abhijithvijayan** [comment]
any status on merging this PR?

**Comment by agneym** [comment]
I would have thought this was an important issue because people using Postgres could not even start their demo project without running into an error. But a month after this PR, not so much.

**Comment by kyleoliveiro** [comment]
Was this issue addressed or is Knex still unusable in production?

**Comment by borg** [comment]
New user, followed latest getting started with postgres. Tables are still not being created by adding lists. Have to run 
```
npm run create-tables
```
which will complain unless db dropped. 
Using  "@keystonejs/adapter-knex": "^10.0.1"

In what version should it be resolved?

**Pull request closed without merging**